tests/sympc/tensor/replicatedshare_tensor_test.py

```diff
@@ -318,9 +336,39 @@ def test_ops_public_mul_integer_matrix(get_clients, parties, security):
     SessionManager.setup_mpc(session)
 
     secret = torch.tensor([[-100, 20, 30], [-90, 1000, 1], [1032, -323, 15]])
-    value = torch.tensor([[-1, 2, 3], [-9, 10, 1], [32, -23, 5]])
+    value = 8
 
+    op = getattr(operator, "mul")
     tensor = MPCTensor(secret=secret, session=session)
-    result = tensor * value
+    shares = [op(share, value) for share in tensor.share_ptrs]
+    result = MPCTensor(shares=shares, session=session)
 
     assert (result.reconstruct() == (secret * value)).all()
+
+
+def test_truediv_exception() -> None:
+    rst = ReplicatedSharedTensor(shares=[1])
+    with pytest.raises(ValueError):
+        rst / 1.55
+
+
+def test_truediv() -> None:
+    secret = 10.25
+    rst = ReplicatedSharedTensor(shares=[secret])
+    rst = rst / 2
+    expected_res = rst.fp_encoder.encode(secret) // 2
+    assert rst.shares[0] == expected_res
+
+
+def test_rshift_exception() -> None:
+    rst = ReplicatedSharedTensor(shares=[1])
+    with pytest.raises(ValueError):
+        rst >> 1.55
+
+
+def test_rshift() -> None:
+    secret = 10.25
+    rst = ReplicatedSharedTensor(shares=[secret])
+    rst = rst >> 2
+    expected_res = rst.fp_encoder.encode(secret) >> 2
+    assert rst.shares[0] == expected_res
```

Comment:
Q: Shouldn't we test the scenario where we do reconstruction and the values are "approximately" the same? (also for the `truediv`)?

Reply:
Sure,I will do it :+1:

Reply:
For the MPCTensor it would not work , as we would get the carry bit issue.

Reply:
Dicussed in slack.